src/main/java/org/broadinstitute/hellbender/engine/AssemblyRegion.java

```diff
@@ -588,7 +588,8 @@ public boolean isFinalized() {
 
         // Use the provided activity profile to determine the bounds of each assembly region:
         List<AssemblyRegion> assemblyRegions = new ArrayList<>();
-        for ( final AlignmentContext pileup : locusIterator ) {
+        final Iterable<AlignmentContext> locatableIterable = () -> locusIterator;
+        for ( final AlignmentContext pileup : locatableIterable ) {
```

Comment:
Can you explain why you decided to make `LocusIteratorByState` non-iterable? This does not strike me as an improvement from a client perspective...